File: uploads/projects/1775729196155/check_jar.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

try {
    // 使用7z命令列出jar文件内容
    const output = execSync('7z l secexample-1.0.jar', { encoding: 'utf8' });
    console.log(output);

    // 检查是否包含lib目录
    if (output.includes('lib/')) {
        console.log('\n这是一个fatJar格式的jar文件');

    } else {
        console.log('\n这不是一个fatJar格式的jar文件');
    }
} catch (error) {
    console.error('Error:', error.message);
}

File: uploads/projects/1775729196155/check_jar.py
import zipfile
import os
import sys

jar_path = 'secexample-1.0.jar'

try:
    with zipfile.ZipFile(jar_path, 'r') as z:
        files = z.namelist()
        print(f"Total files: {len(files)}")

        # 查找lib目录
        lib_files = [f for f in files if f.startswith('lib/')]
        print(f"Files in lib/: {len(lib_files)}")

        if lib_files:
            print("\nFirst 20 files in lib/:")
            for f in lib_files[:20]:
                print(f"  {f}")

            print("\nThis is a fatJar format")
        else:
            print("\nNo lib/ directory found - not a fatJar")

        # 显示主要目录
        print("\nMain directories:")
        main_dirs = set()
        for f in files:
            parts = f.split('/')
            if len(parts) > 1:
                main_dirs.add(parts[0])

        for d in sorted(main_dirs):
            print(f"  {d}/")

except Exception as e:
    print(f"Error: {e}")
    sys.exit(1)
